panel/io/state.py

```diff
@@ -553,6 +553,24 @@ def publish(
             self._rest_endpoints[endpoint] = ([parameterized], parameters, cb)
         parameterized.param.watch(cb, parameters)
 
+    def run_callback(self, callback: Callable([], None)) -> None:
+        """
+        Run a synchronous callback immediately or schedule an asynchronous
+        callback on the event loop.
```

Comment:
Something like this:

```suggestion
        Executes both synchronous and asynchronous callbacks appropriately depending on 
        the context the application is running in. When running on the server callbacks are 
        scheduled on the event loop ensuring the Bokeh Document lock is acquired and
        models can be modified directly.
```